main.go

```diff
@@ -66,53 +72,114 @@ const (
 	webhookPort = 9443
 )
 
-func runPrometheusServer(metricsAddr string, tlsOptions common.SSPTLSOptions) error {
-	setupLog.Info("Starting Prometheus metrics endpoint server with TLS")
-	metrics.Registry.MustRegister(common_templates.CommonTemplatesRestored)
-	metrics.Registry.MustRegister(common.SSPOperatorReconcileSucceeded)
-	handler := promhttp.HandlerFor(metrics.Registry, promhttp.HandlerOpts{})
-	mux := http.NewServeMux()
-	mux.Handle("/metrics", handler)
+// This callback executes on each client call returning a new config to be used
+// please be aware that the APIServer is using http keepalive so this is going to
+// be executed only after a while for fresh connections and not on existing ones
+func getConfigForClientCallback(cfg *tls.Config, apiClient client.Client, ctx context.Context) (*tls.Config, error) {
+	var err error
+	var tlsProfile *ocpconfigv1.TLSSecurityProfile
 
-	minTlsVersion, err := tlsOptions.MinTLSVersionId()
-	if err != nil {
-		return err
+	var sspList ssp.SSPList
+	if err := apiClient.List(ctx, &sspList, &client.ListOptions{}); err != nil {
+		return nil, err
 	}
 
-	tlsConfig := tls.Config{
-		CipherSuites: tlsOptions.CipherIDs(&setupLog),
-		MinVersion:   minTlsVersion,
+	if len(sspList.Items) != 0 {
+		tlsProfile = sspList.Items[0].Spec.TLSSecurityProfile
 	}
 
-	server := http.Server{
-		Addr:      metricsAddr,
-		Handler:   mux,
-		TLSConfig: &tlsConfig,
+	if tlsProfile == nil {
+		cfg.MinVersion = crypto.DefaultTLSVersion()
+		cfg.CipherSuites = nil
+		return cfg, nil
+	}
+
+	if tlsProfile.Type == ocpconfigv1.TLSProfileCustomType {
+		cfg.CipherSuites = common.CipherIDs(tlsProfile.Custom.Ciphers)
+		cfg.MinVersion, err = crypto.TLSVersion(string(tlsProfile.Custom.MinTLSVersion))
+		if err != nil {
+			return nil, err
+		}
+		return cfg, nil
+	}
+
+	cipherNames, minTypedTLSVersion := ocpconfigv1.TLSProfiles[tlsProfile.Type].Ciphers, ocpconfigv1.TLSProfiles[tlsProfile.Type].MinTLSVersion
+	cfg.CipherSuites = common.CipherIDs(cipherNames)
+	cfg.MinVersion, err = crypto.TLSVersion(string(minTypedTLSVersion))
+	if err != nil {
+		return nil, err
 	}
+	return cfg, nil
+}
+
+type prometheusServer struct {
+	server   *http.Server
+	client   client.Client
+	certPath string
+	keyPath  string
+}
+
+func (s *prometheusServer) InjectClient(client client.Client) error {
+	s.client = client
+	return nil
+}
+
+func (s *prometheusServer) NeedLeaderElection() bool {
+	return false
```

Comment:
What's the point in always returning false?

Reply:
This interface is used by the manager to check if the `Runnable` should run only when leader election succedes: 
https://github.com/kubevirt/ssp-operator/blob/862a7efa8fb1bb3e1f20494996b45c7bb1479773/vendor/sigs.k8s.io/controller-runtime/pkg/manager/manager.go#L339-L344

In case of metrics server, we want it to start for all the pods, even ones that fail leader election.

Reply:
It indicates that this server does not need leader election. As far as I know, it is required by the manager. Maybe @akrejcir can elaborate more.

Reply:
ACK

Reply:
Might add a comment about this interface?

Reply:
Can you check if this interface is also deprecated?

Reply:
This interface should not be deprecated.